gemini/src/main/java/com/techempower/gemini/path/MethodSegmentHandler.java

```diff
@@ -550,20 +556,71 @@ protected String getAccessControlAllowMethods(PathSegments segments, C context)
   /**
    * Details of an annotated path segment method.
    */
-  protected static class PathSegmentMethod
+  protected static class PathSegmentMethod extends BasicPathHandlerMethod
   {
     public final String name;      // For debugging purposes only.
-    public final HttpMethod httpMethod;
     public final int index;
     public final boolean contextParameter;
     
-    public PathSegmentMethod(String name, HttpMethod httpMethod, int index, 
-        boolean contextParameter)
+    public PathSegmentMethod(Method method, HttpMethod httpMethod,
+        MethodAccess methodAccess)
     {
-      this.name = name;
-      this.httpMethod = httpMethod;
-      this.index = index;
-      this.contextParameter = contextParameter;
+      super(method, httpMethod);
+
+      this.name = method.getName();
+
+      final Class<?>[] parameterTypes = method.getParameterTypes();
+      this.index = methodAccess.getIndex(method.getName(), parameterTypes);
+
+      if (parameterTypes.length == 0)
+      {
+        // No arguments, which means no context or body parameter. If this
+        // incorrectly has a @Body parameter then the base class will have
+        // caught that by now.
+        this.contextParameter = false;
+      }
+      else if (parameterTypes.length == 1)
+      {
+        // One parameter could be either a context or body parameter.
+        // Check the parameter type to find out.
+        // TODO should this be something like parameterTypes[0].isAssignableFrom(Context.class)?
+        //      If so, make sure to adjust the below checks as well.
```

Comment:
Interested in your thoughts on this question. Gemini supports a Context subclass and allows handlers to specify this type, e.g. `public class MyHandler extends MethodUriHandler<MyContext>`. If a handler method takes in `MyContext` then I don't this this check will work. However I left it because this is what was done before.

Reply:
Leave as is for now - Brian will fix if he wants.